### PR TITLE
Update 2.1.0

### PR DESCRIPTION
**Pull request by mjadach-iv**
Changelog:
 - send message modal - order of peers: me, aliases (sorted by aliases), peers (sorted by peersIds)
 - rename enviorement to network and move it to the network section in Info subpage
 - add Block number to Info subpagte
 - add Block checksum to Info subpagte
 - change send message tag from 1 to 4677
 - remove usused code related to Staking Hub
 - change version to reflect compatible major and minor version of HOPRd

image

## Summary by CodeRabbit

- **Bug Fixes**
  - Updated `SendMessageModal` to use the correct `tag` value in message payloads (from `1` to `4677`).

- **New Features**
  - Added automatic sorting of addresses in the `SendMessageModal` for improved organization.

- **Improvements**
  - Enhanced data extraction logic to include `checksum`, `blockNumber`, and `nodeStartEpoch` for better metrics tracking.
  
- **Refactor**
  - Streamlined the retrieval of peers and addresses for more efficient processing in `SendMessageModal`.
  
- **Chores**
  - Added new fields (`nodeStartEpoch`, `checksum`, `blockNumber`) to the initial state configuration for more comprehensive state management.

### CODE REVIEW

File: src/components/Modal/node/SendMessageModal.tsx [hunk 1]
```diff
@@ -67,6 +75,28 @@ type SendMessageModalProps = {
   tooltip?: JSX.Element | string;
 };
 
+function sortAddresses(
```
Comment (esterlus) [inline src/components/Modal/node/SendMessageModal.tsx:78]: would be good to document the order you are trying to achieve here.

File: src/components/Modal/node/SendMessageModal.tsx [hunk 2]
```diff
@@ -67,6 +75,28 @@ type SendMessageModalProps = {
...
+  me: AddressesType,
+  peerIdToAliasLink: {
+    [peerId: string]: string
+  }
+) : string[] {
+  if(!peers || !me) return [];
+  const connectedPeers = peers.connected;
+  const myAddress = me.hopr as string;
+  const peerIdsWithAliases = Object.keys(peerIdToAliasLink);
+  if(peerIdsWithAliases.length === 0) return [myAddress, ...connectedPeers.map(peer => peer.peerId)];
```
Comment (esterlus) [inline src/components/Modal/node/SendMessageModal.tsx:89]: you are not checking if connected peers contain `myAddress`.
This is something that you would do if you have peerIdsWithAliases (see line 95)

File: src/components/Modal/node/SendMessageModal.tsx [hunk 3]
```diff
@@ -67,6 +75,28 @@ type SendMessageModalProps = {
...
+  let array = [myAddress];
+  peerIdsWithAliases.forEach(peerId => {
+    if(peerId !== myAddress) array.push(peerId)
+  });
+  connectedPeers.forEach(peer => {
+    if(peer.peerId !== myAddress && !peerIdsWithAliases.includes(peer.peerId) ) array.push(peer.peerId)
```
Comment (esterlus) [inline src/components/Modal/node/SendMessageModal.tsx:95]: includes on an array can be expensive with larger sizes arrays. consider converting into a set before checking includes

**Review by esterlus — CHANGES_REQUESTED** [review]
Please fix minor logic misalignment in sortAddresses

File: src/components/Modal/node/SendMessageModal.tsx (hunk 1 above)
Comment (mjadach-iv) [reply src/components/Modal/node/SendMessageModal.tsx:78]: good idea

File: src/components/Modal/node/SendMessageModal.tsx (hunk 3 above)
Comment (mjadach-iv) [reply src/components/Modal/node/SendMessageModal.tsx]: Used the peerIdToAliasLink which I already have. Good point

File: src/components/Modal/node/SendMessageModal.tsx (hunk 2 above)
Comment (mjadach-iv) [reply src/components/Modal/node/SendMessageModal.tsx]: Re-did whole thing

**Review by esterlus — APPROVED** [review]

**Pull request merged**